%!TEX root = ../Main.tex

\chapter{Genetic map of \textit{Yr15} with RNA-Seq}
\chaptermark{Genetic map of \textit{Yr15}}
\label{yr15}
%This section describes in detail than the paper of \citet{Ramirez-Gonzalez-2014}
 
%Breeding importance of \textit{Yr15} and original source (an introgression of \textit{T. diccocoides}). 
Wheat breeding programs aim to improve the wheat lines available for production.
One way to facilitate the selection of elite lines is to find molecular markers that can facilitate the selection of seeds without having to wait until the phenotype is observed.   
Bulk Segregant Analysis (BSA) consists on pooling the DNA of individuals with contrasting phenotypes \citep{Michelmore1991} on a segregating population. 
The bulks show as heterozygous except for the region that is linked to the trait of interest. 
This approach can be used to identify SNPs using High Throughput Sequencing, such as: exome capture \citep{Hodges2007}, RNA-Seq \citep{Pickrell2010}, whole genome resquencing \citep{Schneeberger2009}, among others. 

One of the traits desired in an elite line is the resistance to pathogens, such as \textit{Puccinia striiformis} f. sp.  \textit{tritici}, the fungi responsible of yellow rust.
A source of resistance genes is are introgressions from other species, suchas \textit{Triticum diccocides}. 
In the University of Sydney a collection of Near Isogenic Lines (NILs) with introgressions to several Yellow Rust resistance genes on a susceptible background were developed \citep{Wellings1998}. 
On this chapter the NIL for the \textit{Yr15} locus is used to produce a mapping population to improve diagnostic markers. 
The population was sequenced using RNA-Seq and a bioinformatic pipeline was developed to score Single Nucleotide Polymorphisms (SNPs) linked to the \textit{Yr15} locus.  
Finally, the best candidate SNPs where selected to produce a genetic map which lead to a triplet of markers diagnostic to the target locus. The steps described in this chapter were first published in \citet{Ramirez-Gonzalez2015c} and the results of this chapter are published in \citet{Ramirez-Gonzalez2015b}.

\section{Mapping population}


\begin{figure}
%\begin{wrapfigure}[17]{R!}{7cm}
    \centering
     
     \begin{subfigure}[b]{0.4\textwidth}
        \includegraphics[width=1\textwidth]{Yr15/Figures/population/Yr15Photo.png}
        \caption{}
        \label{fig:yr15.yr15Photo}
    \end{subfigure}
    ~
    \begin{subfigure}[b]{0.4\textwidth}
        \includegraphics[width=1\textwidth]{Yr15/Figures/population/AVSPhoto.png}
        \caption{}
        \label{fig:yr15:avsPhoto}
    \end{subfigure}

     \begin{subfigure}[b]{0.8\textwidth}
        \includegraphics[width=1\textwidth]{Yr15/Figures/population/F2Population.pdf} 
    \caption{ }
    \label{fig:yr15:f2}
	\end{subfigure}

    \caption{Response of (\subref{fig:yr15.yr15Photo}) Avocet + \textit{Yr15} and (\subref{fig:yr15:avsPhoto}) Avocet when inoculated with \textit{Puccinia striiformis} f. sp.  \textit{tritici} at the three leaf stage. (\subref{fig:yr15:f2}) The phenotype of the $F_{2}$ population was used to produce 6 bulks, 3 resistant and 2 susceptible. The RNA was pooled in bulks accordingly. }

%\end{wrapfigure}
\end{figure}

The population was developed by crossing the resistant line `Avocet + \textit{Yr15}` (\textit{Yr15}) \citep{Wellings1998}, Figure \ref{fig:yr15.yr15Photo}, to the susceptible line Avocet (AVS), Figure \ref{fig:yr15:avsPhoto}. 
$F_{2}$ seeds from tree independent $F_{1}$ plants where sown and tissue was collected, before the fungal inoculation to avoid the effect of the response on the gene expression.  
The plants were challenged at the three leaf stage as it is know that \textit{Yr15} confers resistance in seedlings \citep{Gerechter-Amitai1989}.
The expected segregation on an $F_{2}$ population is 3:1 (resistant:susceptible), since \textit{Yr15} is a dominant gene.
From the 232 plants in the $F_{2}$ population that germinated, 187 were resistant and 45 were susceptible, which deviates slightly from the expected ratio ($\chi^{2}=0.049$).
Segregation distortion has been shown for the same \textit{Yr15} donnor \citep{Randhawa2009}, however the decresed number of succeptible plants can be explained by escapes in the virulence essays (i.e. plants scored as resistant without the \textit{Yr15} locus).   For this study we extracted DNA from individual plants in the $F_{2}$ population and we bulked RNA on 6 different bulks: 3 resistant and, 3 succeptible, Figure \ref{fig:yr15:f2}. 

\section{Sequencing and mapping} 
The sequencing of the bulks and the parents were done on 4 lanes of Illumina Hi-Seq2000. 
Each one of the progenitors was sequenced on a single lane, to have enough coverage to call for SNPs


In the experiment, each of the parentals where sequxÂ§enced in a single Illumina Hi-Seq2000 Lane and the six bulks were barcoded to be sequenced on two lanes. The yield of the sequencing is the following: 

\begin{center}
\begin{tabular}{rrccccc}
\toprule
Library & name & Bar code & Lane   &  Reads (\e{8} bp)\\ 
\midrule
LIB1715 & Bulk R1& ATCACG & 1 	& 0.77\\
LIB1716 & Bulk R2& TAGCTT & 1 		& 1.20\\
LIB1717 & Bulk R3& ACTTGA & 2 	& 0.96  \\ 
LIB1718 & Bulk S1& GGCTAC & 2 	& 1.64   \\ 
LIB1719 & Bulk S2 & CGTACG & 2 	& 1.49  \\ 
LIB1720 & Bulk S3 & GTGGCC & 1 	&1.88  \\ 
LIB1721 & AvocetS & N/A & 3 		& 4.13 \\ 
LIB1722 & AvocetS + \textit{Yr15} & N/A & 4 	& 3.99  \\ 

\bottomrule
\end{tabular}
\end{center}



% \section{Sequencing Quality Control}

% To ensure that the quality of the sequencencing was good, fastqc-0.10  was run with its default parameters in each one of the fastq files.  The GC content was around 52\% in all the samples (Appendx \ref{App:AppendixQCGC}), which is expected as the sample should be of coding regions, and for wheat the reported GC content in genes is around 55\%.  The quality of the reads is fairly consistent, in general dropping after the base 80 across the samples (Appendix \ref{App:AppendixQCRead}). 

% As part of the Sequence Informatics analysis, we screened for some of the common contaminants and one of the samples seemed to have several reads with common 21-kmers with Xanthomonas Campestris. Assembling the reads containing kmers which seemed contaminated resulted on 16S and 23S fragments, which are very likely that it is a conserved motif. Still, w went further and looked for Wheat's 16S in NCBI, run an alignment and I got contamination  (Table \ref{tab:aln16S}).

% \begin{table}[h!]

% \begin{center}

% \begin{tabular}{rcccccc}
% \toprule
% Library 				&  Total reads  	& 16S percentage\\ \midrule
%  Bulk R1		&  7.68\e{7} 	& 8.57\% \\ 
%  Bulk R2		&  1.20\e{8}   	& 0.39\% \\ 
%  Bulk R3		&  9.58\e{7}  	& 1.23\% \\ 
%  Bulk S1		& 1.64\e{8}   	& 0.31\% \\ 
%  Bulk S2 		& 1.49\e{8}  	& 4.10\%\\ 
%  Bulk S3  		&1.88\e{8}  	& 0.28\%\\ 
% AvocetS 		& 4.13\e{8} 	& 0.37\%\\ 
%  AvocetS(Yr15) & 3.99\e{8}	 &0.50\%\\ 

% \bottomrule
% \end{tabular}
% \end{center} 
% \caption{Amount of sequence aligning to wheat 16S. Bulks R1 and S2 are have an overrepresentation of 16S}
% \label{tab:aln16S}
% \end{table}



% \subsection{Alignments to Unigenes v60 with BWA}


% All the samples where aligned to the Unigenes v60 (56,954 genes) using BWA 0.5.10\cite{Li2009} (Table \ref{tab:alnUv60}). The alignment provided showed that a few genes were overly expressed (Figure  \ref{fig:Uv60cumExp}), however we still have have over 20,000 genes with a coverage greater than 20x in all the samples (\ref{fig:Uv60sortCov}). The Bulk R1 shows less genes with a coverage above the threshold, which is expected as it is the sample with a higher representation of 16S  (Table \ref{tab:aln16S})

% \begin{table}[h!]

% \begin{center}
% \begin{tabular}{rccccc}
% \toprule
% Sample	& Total reads	& Mapped reads	& \%Mapped reads	& Proper pairs & \% Proper pairs	\\
% \midrule 
% Bulk R1	& 7.68E+07	& 5.02E+07	& 65.38\%	& 4.37E+07	& 56.93\% \\
% Bulk R2	& 1.20E+08	& 7.85E+07	& 65.38\%	& 6.67E+07	& 55.57\% \\
% Bulk R3	& 9.58E+07	& 6.08E+07	& 63.52\%	& 5.09E+07	& 53.18\% \\
% Bulk S1	& 1.64E+08	& 1.02E+08	& 62.36\%	& 8.47E+07	& 51.79\% \\
% Bulk S2	& 1.49E+08	& 9.47E+07	& 63.55\%	& 8.00E+07	& 53.64\% \\
% Bulk  S3	& 1.88E+08	& 1.21E+08	& 64.65\%	& 1.02E+08	& 54.38\% \\
% AvocetS	& 4.13E+08	& 2.59E+08	& 62.68\%	& 2.13E+08	& 51.53\% \\
% AvocetS(Yr15)	& 3.99E+08	& 2.65E+08	& 66.33\%	& 2.20E+08	& 55.24\% \\
% \bottomrule
%  \end{tabular}

% The following plot shows the effect of not having a normalised sample, where less than 1,000 genes get around half of the sample.
% \caption{Count and percentage of the samples aligned }
% \label{tab:alnUv60}
% \end{center}
% \end{table}
% \begin{figure}
%         \centering
%         \begin{subfigure}[b]{0.80\textwidth}
%                 \centering
%                 \includegraphics[width=\textwidth]{Results/images/CumulativeExpression.png}
%                 \caption{Cumulative expression of genes. The horizontal line represents half of the sample }
%                 \label{fig:Uv60cumExp}
%         \end{subfigure}
        
%         \begin{subfigure}[b]{0.80\textwidth}
%                 \centering
%                 \includegraphics[width=\textwidth]{Results/images/sorted_coverage.png}
%                 \caption{Coverage of genes, sorted by coverage}
%                 \label{fig:Uv60sortCov}
%         \end{subfigure}
%         \caption{Volume of sequence assigned to each gene. Few genes are over expressed, but we still have enough coverage (above 20x) to continue the analysis. }
%         \label{fig:Uv60expr}
% \end{figure}

%RNA-Seq and the decision to call SNPs on gene models rather than the whole reference.  Details of the mapping against the Wheat UniGenes \cite{PontiusJUWagnerL2002} and the UCW. \cite{Krasileva2013} gene models.  

\section{SNP Calling}. 
\verb|Ruby| implementation of the methodology described by \citet{Trick2012}. 

\section{Bulk Frequency Ratios} 
Results of the simple SNP calls from the progenitors and how the score of the Bulk Frequency Ratios(BFR) improve the location of the SNPs. 

\section{\textit{In silico} mapping}
Mapping of the gene models to the IWGSC CSS \cite{Mayer2014} reference and the location of the SNPs using the genetic map from \citet{Wang2014}.

\section{Assay selection}. 
The selection criteria to decide which SNPs where selected to produce the genetic map: BFR$>$6, in the short arm of chromosome group 1 and from the \textit{Yr15} progenitor.

\section{Genetic map} 
\label{yr15:geneticMap}
The three versions of the genetic map: With a subset of the F\textsubscript{2} population

\section{Assembly of the transcriptome} 
A comparison between thef known unigenes and the transcript from the progenitors. Since \textit{Yr15} comes from an introgression with \textit{T. diccocoides}, some novel transcripts can be extracted. Analysis of the gels from Mitaly? 

\section{Conclusions} 
Remarks on how this techinque can be used to do fine-mapping and that if I were to start the project now I would  use exome capture or Ren-Seq. 