%!TEX root = ../Main.tex

\chapter{Genetic map of \textit{Yr15} with RNA-Seq}
\chaptermark{Genetic map of \textit{Yr15}}
\label{yr15}
%This section describes in detail than the paper of \citet{Ramirez-Gonzalez-2014}
 
%Breeding importance of \textit{Yr15} and original source (an introgression of \textit{T. diccocoides}). 
Wheat breeding programs aim to improve the wheat lines available for production.
One of the traits desired in an elite line is the resistance to pathogens, such as \textit{Puccinia striiformis} f. sp.  \textit{tritici}, the fungi responsible of yellow rust.
A source of resistance genes is are introgressions from other species, such as \textit{Triticum diccocides}. 
In the University of Sydney a collection of \glspl{nil} with introgressions to several Yellow Rust resistance genes on a susceptible background were developed \citep{Wellings1998}. 
On this chapter the NIL for the \textit{Yr15} locus is used to produce a mapping population to improve diagnostic markers. 

%TODO: Paragraph explaining NILs


Line selection can be done with molecular markers that can be used to test if certain allele is present in a line, without the need to do a phenotype.
To find which regions are linked to a trait the use of $F_{2}$ mapping populations is a common practice.
The population is produced by crossing two homozygous parents ($P_1$ and $P_{2}$) with different alleles, A/A (dominant) and a/a (recessive).
When the trait is dominant and has a mendelian segregation, the $F_1$ population show the dominant trait, as it has a copy of each allele (A/a). 
The $F_1$ is then self-crossed to and the population segregates with a ration 1:2:1, dominant:heterozygous:recessive respectively.
This generates a population with a phenotype ratio of 3:1 (dominant:recessive), since the effect of the recessive allele is masked by the dominant gene (\citealt{VanOoijen2013}; Figure \ref{fig:yr15:f2schematic}).  

\begin{SCfigure}
  \centering
    \includegraphics[width=0.4\textwidth]{Yr15/Figures/population/F2schematic.pdf}
  \caption{The cross of two homozygous parents, $P_{1}$ and $P_{2}$, with a dominant and a recessive allele of a gene produce an heterozygous $F_{1}$. The $F_{1}$ crossed with itself produce a segregating $F_{2}$ population with a 1:2:1 ratio (A/A:A/a:a/a). The upper and lower cases represent dominant and recessive alleles  }. 
  \label{fig:yr15:f2schematic}
\end{SCfigure}


\gls{bsa} consists on pooling the DNA of individuals with contrasting phenotypes \citep{Michelmore1991} on a segregating population. 
The bulks show as heterozygous except for the region that is linked to the trait of interest. 
This approach can be used to identify SNPs using High Throughput Sequencing, such as: exome capture \citep{Hodges2007}, RNA-Seq \citep{Pickrell2010}, whole genome resquencing \citep{Schneeberger2009}, among others. 


\begin{figure}
\includegraphics[width=1\textwidth]{Yr15/Figures/bfr.pdf}
\caption{Illustration of a non-informative homoeologous SNP (G181T) present in both parental lines, and an informative allelic SNP (G184A), only present in the resistant progenitor Avocet S + Yr15. The consensus sequences from the parental genotypes include this information in the form of ambiguity codes (K and R, respectively). In the bulks, the individual reads align across the reference sequence, with matches indicated by dots, and polymorphisms at positions 181 and 184 indicated by the corresponding nucleotide variants at those positions. The SNP index is calculated as the frequency of the informative allelic SNP in each bulk. The Bulk Frequency Ratio is the quotient of the resistant and susceptible bulk SNP Indexes. Figure previously published in \citet{Ramirez-Gonzalez2015c}. }
\label{fig:yr15:bfr}
\end{figure}


To Call for SNPs from RNA-Seq a reference transcriptome is used as target when aligning the reads. 
The \gls{bfr} methodology can work on organisms has more than one pseudo genome and that the genes are not necessarily fully characterised independently among homoeologues or paralogues, you can have in a single reference collapsing similar regions. 
The UniGenes database, from NCBI, contains the genes of each species, with all the variations of each gene automatically collapsed and represented as with the longest \acrshort{cdna} \citep{PontiusJUWagnerL2002}. 
The \acrshort{ucw}  genes described in \citet{Krasileva2013} contains 94,177 models from tetraploid and hexaploid wheat, assembled and phased to separate different homoeologues. 
Both gene sets are complement each other, however, the \acrshort{ucw} gene models should provide an improved alignment, since the different homoeologues aren't merged in a single model, a possible side effect of the UniGene pipeline. 

Homoeologous variants, as exemplified by the G$>$T variant at position 181; K in consensus (Figure \ref{fig:yr15:bfr}), will produce the same ambiguity code for both parental consensus sequences and can therefore be excluded. 
Real allelic SNPs between the parental genotypes, exemplified by the G$>$A variant at position 184; R in consensus, are distinguished by the presence in one, but not the other parental consensus sequence. 
The allelic SNPs are then examined further with the alignments of the bulks to identified the SNPs that are enriched on the resistant plants.
The SNP index is the proportion of times an alternative allele is observed over the coverage at certain, in the example the the susceptible bulk has an SNP index of $1/8=0.125$ and $6/8=0.75$ for the resistant bulk \citep{Takagi2013a}. traditional
The \acrshort{bfr} are then calculated by dividing the SNP Index of sample containing the target phenotype (resistance) over the sample without the trait (susceptible), on the example is $0.75/0.125=6$.  
A high BFR suggests that the \acrshort{snp} is linked to the target trait \citep{Trick2012}. 


Finally, the best candidate SNPs where selected to produce a genetic map which lead to a triplet of markers diagnostic to the target locus. 

The steps described in this chapter were first published in \citet{Ramirez-Gonzalez2015c} and the results of this chapter are published in \citet{Ramirez-Gonzalez2015b}.

\section{Mapping population}


\begin{figure}
%\begin{wrapfigure}[17]{R!}{7cm}
    \centering
     
     \begin{subfigure}[b]{0.4\textwidth}
        \caption{}
        \includegraphics[width=1\textwidth]{Yr15/Figures/population/Yr15Photo.png}
        \label{fig:yr15.yr15Photo}
    \end{subfigure}
    ~
    \begin{subfigure}[b]{0.4\textwidth}
        \caption{}
        \includegraphics[width=1\textwidth]{Yr15/Figures/population/AVSPhoto.png}
        \label{fig:yr15:avsPhoto}
    \end{subfigure}

     \begin{subfigure}[b]{0.8\textwidth}
     \caption{}
        \includegraphics[width=1\textwidth]{Yr15/Figures/population/F2Population.pdf} 
    \label{fig:yr15:f2}
	\end{subfigure}

    \caption{Response of (\subref{fig:yr15.yr15Photo}) Avocet + \textit{Yr15} and (\subref{fig:yr15:avsPhoto}) Avocet when inoculated with \textit{Puccinia striiformis} f. sp.  \textit{tritici} at the three leaf stage. (\subref{fig:yr15:f2}) The phenotype of the $F_{2}$ population was used to produce 6 bulks, 3 resistant and 2 susceptible. The RNA was pooled in bulks accordingly. Adapted from \citep{Ramirez-Gonzalez2015b}}

%\end{wrapfigure}
\end{figure}

The population was developed by crossing the resistant line \gls{yr15} \citep{Wellings1998}, Figure \ref{fig:yr15.yr15Photo}, to the susceptible line \gls{avs}, Figure \ref{fig:yr15:avsPhoto}. 
\acrshort{yr15} is a \gls{nil} of a 6th generation \gls{bc} and the \acrshort{avs} background is highly succeptible to yellow rust, hence the resistance is coffered by the \acrshort{yr15} locus. 
$F_{2}$ seeds from tree independent $F_{1}$ plants where sown and tissue was collected, before the fungal inoculation to avoid the effect of the response on the gene expression.  
The plants were challenged at the three leaf stage as it is know that \textit{Yr15} confers resistance in seedlings \citep{Gerechter-Amitai1989}.
The expected segregation on an $F_{2}$ population is 3:1 (resistant:susceptible), since \textit{Yr15} is a dominant gene.
From the 232 plants in the $F_{2}$ population that germinated, 187 were resistant and 45 were susceptible, which deviates slightly from the expected ratio ($\chi^{2}=0.049$).
Segregation distortion has been shown for the same \textit{Yr15} donnor \citep{Randhawa2009}, however the decresed number of succeptible plants can be explained by escapes in the virulence essays (i.e. plants scored as resistant without the \textit{Yr15} locus).   For this study we extracted DNA from individual plants in the $F_{2}$ population and we bulked RNA on 6 different bulks: 3 resistant and, 3 succeptible ( Figure \ref{fig:yr15:f2}). 

\section{Sequencing and mapping} 

\begin{table}
\centering
\caption{Arrangement and number of sequenced base pairs per sample. }
\label{tab:yr15:reads}
\begin{tabular}{rrccccc}
\toprule
Library & name & Bar code & Lane   &  Reads (\e{8} bp)\\ 
\midrule
LIB1715 & Bulk R1 & ATCACG & 1 	& 0.77\\
LIB1716 & Bulk R2 & TAGCTT & 1 		& 1.20\\
LIB1717 & Bulk R3 & ACTTGA & 2 	& 0.96  \\ 
LIB1718 & Bulk S1 & GGCTAC & 2 	& 1.64   \\ 
LIB1719 & Bulk S2 & CGTACG & 2 	& 1.49  \\ 
LIB1720 & Bulk S3 & GTGGCC & 1 	&1.88  \\ 
LIB1721 & AvocetS & N/A & 3 		& 4.13 \\ 
LIB1722 & AvocetS + \textit{Yr15} & N/A & 4 	& 3.99  \\ 
\bottomrule
\end{tabular}
\end{table}


RNA-Seq was used to avoid sequencing the non-coding regions and reduce the search space.  
The sequencing of the bulks and the parents were done on a single Illumina Hi-Seq2000 each.
The bulks were multiplexed and sequenced on a third of a lane each, as shown on Table \ref{tab:yr15:reads}. 
To ensure that the quality of the sequencencing was good, \verb|fastqc-0.10| \citep{fastqc}  was run with its default parameters in each one of the fastq files.  
The GC content was around 52\% in all the samples (Appendx \ref{App:AppendixQCGC}), which is expected as the sample should be of coding regions, and for wheat the reported GC content in genes is around 55\%.  
The quality of the reads is fairly consistent, in general dropping after the base 80 across the samples (Appendix \ref{App:AppendixQCRead}). 


\input{Yr15/SupplementalTables/alignedCoverage}


\begin{figure}
\centering
\begin{subfigure}{0.29\textwidth}
    \caption{}
     \includegraphics[width=1\textwidth]{Yr15/Figures/CoveragePerGene.pdf} 
    \label{fig:yr15:covPerGene}
\end{subfigure}
~
\begin{subfigure}{0.5\textwidth}
    \caption{}
    \includegraphics[width=1\textwidth]{Yr15/Figures/PercentageOfSnps.pdf} 
    
    \label{fig:yr15:SNPper}
\end{subfigure}

\caption{(\subref{fig:yr15:covPerGene}) Box plot distribution of the gene coverage of the parent reads (\acrshort{avs} and \acrshort{yr15}) across the UCW (blue) and the UniGene (red) gene models. The dashed line represents the 209 minimum coverage required for SNP calling. The full line represents the average coverage across all gene models. (\subref{fig:yr15:SNPper}) Percentage of genes exhibiting SNPs across references. The number of \acrshort{snp}s between the parent reads and the corresponding references was calculated (per 100 bp, rounded). The ‘between-parents’ category corresponds to putative SNPs when comparing the consensus sequence between \acrshort{avs} and \acrshort{yr15} Adapted from \citet{Ramirez-Gonzalez2015b} }
\end{figure}


When the analysis was started, the draft genome and the corresponding annotation where not not release yet, hence gene models where used. 
All the samples where aligned to the Unigenes v60 (56,954 genes) and the gene models from UCW \citep{Krasileva2013} using \verb|BWA 0.5.9| \citep{Li2009}. 
The alignment provided showed that a few genes were overly expressed, however we still have have 22,107 and 36,808 genes, on the Unigenes and the UCW gene set respectivley,  with a coverage greater than 20x in the progenitor with \textit{Yr15}. 
Both gene sets performed similarly in terms of the percentage of genes with reads and percentage of aligned reads. 
For \gls{avs} and \gls{yr15}, the percentage of genes with a coverage of at least $20x$ is $45\%$ and $39\%$ respetively across both references (Figure \ref{fig:yr15:covPerGene}).
Since each individual bulk has a lower coverage, the susceptible and resistant reads were merged \textit{in silico} as: (i) susceptible bulks 1 with 2 (S1 + S2) and resistant bulks 1 with 2 (R1 + R2) and (ii) all the susceptible (S1 + S2 + S3) and resistant bulks (R1 + R2 + R3). 
The merged samples increased the percentage of genes with coverage over 20x  to 44\% and 50\% in the resistant and susceptible bulks (Table \ref{app:seqAlnCov}), which is close to the coverage from the progenitors.

\section{SNP Calling}

The \acrshort{snp} calling was done on positions with a coverage of at least $20x$ on the progenitor lines against the gene reference. The \acrshort{avs} progenitor had roughly $3\%$ more genes with polymorphisms than \acrshort{yr15}, consistent with the difference in coverage, suggesting that with a higher coverage we could recover more \acrshort{snp}s from \acrshort{yr15}.
The UniGenes have a higher number of \acrshort{snp}s because the \acrshort{ucw} gene models have a higher number of monomorphic genes when compared to the UniGenes. (Figure {\ref{fig:yr15:SNPper}}; Table \ref{app:yr15:cntSNP100bp}). 
The difference in the number of relative monomorphic SNPs between references can be explained by the fact that the UniGenes have homoeologues can be represented as a single sequence, as opposed to the UCW set which are homoeologue-specific, improving the mapping to the correct homoeologue in the genes from the UCW set over the UniGenes.

\input{Yr15/SupplementalTables/snpsPer100bp}

Both gene sets were done from varieties different to \acrshort{avs} and are likely to be incomplete, hence we set a low threshold of at least 20\% of the observed nucleotides on any position to call an \acrshort{snp}. 
To represent cases were more than one consensus base is called we use \gls{iuapc} codes (\citet{Cornish-Bowden1985}; Section \ref{lit:ambiguity}; Figure \ref{fig:yr15:bfr}).  
To focus the analysis on informative \acrshort{snp}s, the common varietal SNPs and variations between homoeologues were removed by finding the cases when the consensus call on both progenitors is the same. 
The \acrshort{snp}s that are unique to a single parental were examined in detail. 
There are 66,426 putative SNPs across 16,022 (17\%) \acrshort{ucw} genes and 52,262 \acrshort{snp}s on 11,056 UniGenes (19.4\%; Figure \ref{fig:yr15:geneCount}).  

\begin{SCfigure}
    %\centering
    \includegraphics[width=0.4\textwidth]{Yr15/Figures/geneCounts.pdf} 
    \caption{Gene models with putative SNPs in common between the UCW and UniGenes reference. The intersection represnts the genes that are common in both sets. Adapted from \citet{Ramirez-Gonzalez2015b}}
    \label{fig:yr15:geneCount}
\end{SCfigure}

\input{Yr15/SupplementalTables/genesAssignedtoArm}

The high number of genes with \acrshort{snp}s was unexpected as a \acrshort{bc}6 \acrshort{nil} used for an $F_2$ mapping population expects to have $<1\%$ of the genetic background segregating. 
The both sets of gene models were aligned with BLAT \citep{Kent2002} to the Chinese Spring Chromosome arm survey sequence (CSS; \citealt{Mayer2014}); the alignment resulted on 80,031 (85.0\%) UCW gene models and 41,118 (72.2\%) UniGenes assigned to a chromosome arm (Table \ref{tab:yr15:genesToCSS}). 
The SNPs found in the mapped genes are evenly distributed across all the chromosomes (Figure \ref{fig:yr15:bfr0}), suggesting that the \acrlong{avs} (JIC, UK) used as parent in the $F_{2}$ is different to the \acrlong{avs} used for the \acrshort{yr15} \acrshort{nil} development (University of Sydney, Australia).  


To confirm that the \acrlong{avs} seed stocks from JIC are distinct to the stocks in Sydney  DNA from both stocks was procured and compared with the iSelect 90k wheat SNP chip. 
Between two independent \acrlong{avs} seeds from JIC only 58 out of 71,972 (0.08\%) valid assays were polymorphic. 
Nonetheless, ther are over 5,000 ($>7.5\%$) assays with polymorphisms between  JIC-\acrlong{avs} and \acrlong{avs} from Sydney. 
The different was not expected originally, but considering that the \acrlong{avs} seeds are coming from different stocks and the fact that in both countries commercial varieties with the same name had been released, it is not surprising. 


\section{Bulk Frequency Ratios}

The objective was find the \acrshort{snp}s enriched on each bulk and hence linked to the phenotype, variations from \acrshort{yr15} to resistance and from \acrshort{avs} to susceptibility in the segregating population. 
Across individual bulks, it was possible to score between 15,261 (24.5\%) to 31,891(48.0\%) \acrshort{snp}s across both reference sets.
On the \textit{in silico} mixes over $95\%$ of SNPs where scored (Table \ref{app:yr15:scoredSNPs}), suggesting that the coverage of individual bulks is not enough to score all the SNPs.  
The scoring was done with the \acrlong{bfr} (\citealt{Trick2012};Figure \ref{fig:yr15:bfr}; Section \ref{yr15:sub:bfr}), which has a value that increases as the \acrshort{yr15} allele is observed more times relatively to the \acrshort{avs} allele.


\input{Yr15/SupplementalTables/scoredSNPs}

When increasing the minimum BFR threshold, enrichment of SNPs was observed in the short arm of the group 1 chromosomes (1S). 
Without taking in account the BFR, $~3.6\%$ of the SNPs are located in the 1S group, similar to the number of SNPs located in other groups \ref{tab:yr15:genesToCSS}. 
However, when increasing the threshold  (between $BFR > 5 $ and $BFR > 7$) the relative number of SNPs in group 1S increases. 
After $BFR>7$ the gains in relative enrichment only improves marginally, but the number of called SNPs is reduced (Table \ref{app:yr15:bfrThresholds}; Figure \ref{fig:yr15:bfrChange}).
For that reason, SNPs with a $BFR>6$ were selected for further validation. 
The method described by \citet{Trick2012} was extended by including cases where there is a complete lack of coverage in one of the samples ($BFR=\infty$), which is an ideal case where the linkage between the SNP and the phenotype is perfect. 
A total of 1,582 SNPs across 1,173 genes had a $BFR>6$.

\input{Yr15/SupplementalTables/bfrThresholds}

\begin{figure}
\includegraphics[width=1\textwidth]{Yr15/Figures/bfrChanges.pdf}
\caption{ Effect of BFR threshold on the number of SNPs across the short arm of chromosome group 1. Figure previously published in \citet{Ramirez-Gonzalez2015b} }
\label{fig:yr15:bfrChange}
\end{figure}

\section{\textit{In silico} mapping}
Mapping of the gene models to the IWGSC CSS \citet{Mayer2014} reference and the location of the SNPs using the genetic map from \citet{Wang2014}.

\begin{figure}
	\centering
	\begin{subfigure}{0.4\textwidth}
	\caption{}
	\label{fig:yr15:bfr0}
	\includegraphics[height=0.55\textheight]{/Users/ramirezr/Documents/public_code/PhdThesis/Yr15/Figures/mapping/snpsBFR0.pdf}
	\end{subfigure}
	~
	\begin{subfigure}{0.45\textwidth}
	\caption{}
	\label{fig:yr15:bfr6}
	\includegraphics[height=0.55\textheight]{/Users/ramirezr/Documents/public_code/PhdThesis/Yr15/Figures/mapping/snpsBFR6.pdf}
	\end{subfigure}
	\caption{Genetic location of genes with SNPs between AVS and Yr15. The colour scale indicates the percentage of genes with SNPs per centi-Morgan (cM) across the 21 wheat chromosomes. The location of the genes was determined by the best alignment to the CSS scaffolds, and the location of these was determined by their position on a genetic map \citep{Wang2014} (\subref{fig:yr15:bfr0}). All the SNPS between progenitors. Note the lack of enrichment across any individual chromosome. (\subref{fig:yr15:bfr6}) SNPs with BFR$>$6. Note the enrichment in Chromosome 1B }
	\label{fig:yr15:bfrs:0-6}
\end{figure}


\section{Assay selection}. 
The selection criteria to decide which SNPs where selected to produce the genetic map: BFR$>$6, in the short arm of chromosome group 1 and from the \textit{Yr15} progenitor.

\section{Genetic map} 
\label{yr15:geneticMap}
The three versions of the genetic map: With a subset of the F\textsubscript{2} population

\section{Bioinformatic methods}

\subsection{Alignment reads to gene models}

The raw output from the Illumina HiSeq 2000 was processed with Casava v1.8 \citep{casavaBCL}. 
Lanes 1 and 2, containing multiplexed bulks (Table \ref{tab:yr15:reads}) was demultiplexed with a tolerance of 1 mismatch in the barcode. 
Lanes 3 and 4 contained the parental sequences without a barcode. 
The FastQ files where left compressed and in chunks of 40,000, as teh default for the BCL conversion pipeline from Casava to allow parallel processing in a cluster environment. 
The quality of the sequencing lanes was assessed with FastQC v0.10.1 \citep{fastqc}. 
The RNA-Seq reads were aligned with BWA 0.5.9 \citep{Li2009} to the wheat UniGene database v60 \citep{PontiusJUWagnerL2002} and to the UCW gene models \citep{Krasileva2013}, including the \textit{T. turgidum} and complementary ORFs \citep{MASWheat2013}.
The alignments where sorted and stored as single BAM files to have random access \citep{Li2009a}. 

%TODO: Snippet with submission of the alignments. 

\subsection{Bulk Frequency Ratios}
\label{yr15:sub:bfr}

%To identify variant candidates, the methodology described in Trick et al. (2012) was used and extended to work with BAM files and to consider cases when a variant is completely absent from the parental sequences. The consensus from the two parents is called, allowing for ambiguities to take into account possible homoeo- logues when at least 20% of the bases differ from the reference. The consensus is stored with standard IUPAC codes (Cornish- Bowden, 1985). On the bases where the consensus is different between the parents, the bulk frequency ratio (BFR) was calculated as in Trick et al. (2012). The algorithm was imple- mented using Ruby on the top of BioRuby (Goto et al., 2010) and extending the functionality of bio-samtools (Ramirez-Gonzalez et al., 2012). The BFRs were calculated independently on each of the three comparisons (Bulk 1: S1–R1, Bulk 2: S2–R2 and Bulk 3: S3–R3) and with in silico mixes of bulks 1 and 2; and bulks 1, 2 and 3 produced by merging the BAM files using samtools (Li et al., 2009). A list was produced with all the scores in a table format containing the BFR, coverage, ratio of the SNP base and the the parent containing the SNP (Data S2–S4).

\lstinputlisting[language=Ruby,firstline=17, lastline=28, caption=Method to find best BLAT alignment]{Yr15/code/find_best_blat_hit.rb}

\lstinputlisting[language=Ruby,firstline=33, lastline=43, caption=Extensions to Bio::Blat::Report::Hit to get the percentage of coverage]{Yr15/code/find_best_blat_hit.rb}
%The UniGenes, UCW gene models and the contigs containing the 46 977 genetically mapped SNPs from Wang et al. (2014) were aligned with BLAT (Kent, 2002) to the CSS scaffolds (IWGSC, 2014). To assign the genes to a specific scaffold, we selected the best hit from the BLAT alignment, provided that at least 60% of the gene was covered in the scaffold with at least 90% of identity. This selection criterion reduces the number of spurious hits in repetitive motifs in the gene and allows the assignment to a homoeologous chromosome even when the correct homoeologue is missing from the CSS scaffolds. The origin of the scaffold was used to assign a putative chromosome arm. In addition, the gene models were also aligned to the cDNA of Hordeum vulgare (International Barley Genome Sequencing Consortium, 2012) from Ensembl! Plants, release 16 (Kersey et al., 2012). The genetic position of the wheat contigs with SNPs was used to calculate the density of SNPs with a BFR > 6 and for all SNPs between AVS and Yr15. After selection using the criteria outlined in the results section, the selected genes were sorted by BFR in the mix between samples to select the top 50 SNPs for further validation via KASP markers.

%\section{Assembly of the transcriptome} 
%A comparison between thef known unigenes and the transcript from the progenitors. Since \textit{Yr15} comes from an introgression with \textit{T. diccocoides}, some novel transcripts can be extracted. Analysis of the gels from Mitaly? 

\section{Discussion} 
Remarks on how this techinque can be used to do fine-mapping and that if I were to start the project now I would  use exome capture or Ren-Seq. 

The references have changed since we started

There are new annotations, now we don't necessarily need to use unigenes anymore. 

Importance of genotyping everything used in an experiment

Mention other people using a similar strategy since this was published. 

We can use different techniques now (exome capture, ren-seq)

The markers are now used by our collaborators. 